wrangler-service/src/main/java/co/cask/wrangler/RequestExtractor.java

```diff
@@ -56,14 +47,15 @@ public boolean hasHeader(String name) {
    * @param defaultValue value to returned if header doesn't exist.
    * @return value defined for the header.
    */
-  public <T> T getHeader(String name, String defaultValue) {
+  public String getHeader(String name, String defaultValue) {
```

Comment:
if it's not annotated as nullable, callers shouldn't pass in null. In this particular method, there should be no reason to call this method passing in null as default, since it's the same as just getting the header.

That being said, there is a place in the code that passes in null. I'll change it to not call this method.